plugin/path_config_rotate_root_test.go

```diff
@@ -0,0 +1,194 @@
+	"time"
+
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-gcp-common/gcputil"
+	"github.com/hashicorp/vault/sdk/logical"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iam/v1"
+)
+
+func TestConfigRotateRootUpdate(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no_configuration", func(t *testing.T) {
+		t.Parallel()
+
+		b, storage := getTestBackend(t)
+		_, err := b.HandleRequest(context.Background(), &logical.Request{
+			Operation: logical.UpdateOperation,
+			Path:      "config/rotate-root",
+			Storage:   storage,
+		})
+		if err == nil {
+			t.Fatal("expected error")
+		}
+		if exp, act := "no configuration", err.Error(); !strings.Contains(act, exp) {
+			t.Errorf("expected %q to contain %q", act, exp)
+		}
+	})
+
+	t.Run("config_with_no_credentials", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.Background()
+		b, storage := getTestBackend(t)
+
+		entry, err := logical.StorageEntryJSON("config", &config{
+			TTL: 5 * time.Minute,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err := storage.Put(ctx, entry); err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = b.HandleRequest(ctx, &logical.Request{
+			Operation: logical.UpdateOperation,
+			Path:      "config/rotate-root",
+			Storage:   storage,
+		})
+		if err == nil {
+			t.Fatal("expected error")
+		}
+		if exp, act := "does not have credentials", err.Error(); !strings.Contains(act, exp) {
+			t.Errorf("expected %q to contain %q", act, exp)
+		}
+	})
+
+	t.Run("config_with_invalid_credentials", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.Background()
+		b, storage := getTestBackend(t)
+
+		entry, err := logical.StorageEntryJSON("config", &config{
+			CredentialsRaw: "baconbaconbacon",
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err := storage.Put(ctx, entry); err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = b.HandleRequest(ctx, &logical.Request{
+			Operation: logical.UpdateOperation,
+			Path:      "config/rotate-root",
+			Storage:   storage,
+		})
+		if err == nil {
+			t.Fatal("expected error")
+		}
+		if exp, act := "credentials are invalid", err.Error(); !strings.Contains(act, exp) {
+			t.Errorf("expected %q to contain %q", act, exp)
+		}
+	})
+
+	t.Run("rotate", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.Background()
+		b, storage := getTestBackend(t)
+
+		// Get user-supplied credentials
+		credsPath := os.Getenv("GOOGLE_CREDENTIALS")
+		credsBytes, err := ioutil.ReadFile(credsPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		creds, err := google.CredentialsFromJSON(ctx, credsBytes, iam.CloudPlatformScope)
+		if err != nil {
+			t.Fatal(err)
+		}
+		parsedCreds, err := gcputil.Credentials(string(credsBytes))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Create http client
+		clientCtx := context.WithValue(ctx, oauth2.HTTPClient, cleanhttp.DefaultClient())
+		client := oauth2.NewClient(clientCtx, creds.TokenSource)
+
+		// Create IAM client
+		iamAdmin, err := iam.New(client)
```

Comment:
Deprecation warning:

> iam.New is deprecated: please use NewService instead. To provide a custom HTTP client, use option.WithHTTPClient. If you are using google.golang.org/api/googleapis/transport.APIKey, use option.WithAPIKey with NewService instead.

This warning hits a few other areas too, but since this is new code it would be good to use the current version.

Reply:
I'm aware. It's not really feasible to update to `NewService` since it directly conflicts with `New`. I'd have to update all the files and change around some internals because of the new constructor. I'm happy to do that, but I'd prefer to take on a "fix all the warnings" PR instead of merging it with this one. Is that okay with you?

Reply:
Sounds good.